packages/ethereum-storage/src/ipfs-manager.ts

```diff
@@ -192,104 +95,32 @@ export default class IpfsManager {
-          });
-
-          // All data has been received
-          res.on('end', () => {
-            let jsonData;
-            try {
-              jsonData = JSON.parse(data);
-            } catch (error) {
-              return reject(
-                new Error('Ipfs object get request response cannot be parsed into JSON format'),
-              );
-            }
-            if (jsonData.Type === 'error') {
-              return reject(new Error(`Ipfs object get failed: ${jsonData.Message}`));
-            }
-            const ipfsDataBuffer = Buffer.from(jsonData.Data, 'base64');
-            const content = this.getContentFromMarshaledData(ipfsDataBuffer);
-            const ipfsSize = ipfsDataBuffer.length;
-            const ipfsLinks = jsonData.Links;
-            resolve({ content, ipfsSize, ipfsLinks });
-          });
-
-          // Error handling
-          res.on('error', (e: string) => {
-            // If maxRetries is set, and we haven't reached maxRetries, retry the request
-            if (
-              this.errorHandlingConfig.maxRetries &&
-              retries <= this.errorHandlingConfig.maxRetries
-            ) {
-              setTimeout(
-                () => resolve(this.read(hash, maxSize, retries + 1)),
-                this.errorHandlingConfig.delayBetweenRetries,
-              );
-            } else {
-              return reject(new IpfsConnectionError(`Ipfs read request response error: ${e}`));
-            }
-          });
-          res.on('aborted', () => {
-            return reject(new IpfsConnectionError('Ipfs read request response has been aborted'));
-          });
-        })
-        .on('timeout', () => {
-          didTimeout = true;
-          // explicitly abort the request
-          getRequest.abort();
-          return reject(new IpfsConnectionError('Ipfs read request timeout'));
-        })
-        .on('abort', () => {
-          return reject(new IpfsConnectionError('Ipfs read request has been aborted'));
-        })
-        .on('error', (e: string) => {
-          // If the error isn't a timeout, maxRetries is set, and we haven't reached maxRetries, retry the request
-          if (
-            !didTimeout &&
-            this.errorHandlingConfig.maxRetries &&
-            retries <= this.errorHandlingConfig.maxRetries
-          ) {
-            setTimeout(
-              () => resolve(this.read(hash, maxSize, retries + 1)),
-              this.errorHandlingConfig.delayBetweenRetries,
-            );
-          } else {
-            return reject(new IpfsConnectionError(`Ipfs read request error: ${e}`));
-          }
-        });
-
-      if (this.ipfsConnection.timeout && this.ipfsConnection.timeout > 0) {
-        getRequest.setTimeout(this.ipfsConnection.timeout);
+    try {
+      if (maxSize !== Number.POSITIVE_INFINITY) {
+        const maxSizeBase64 = ((4 * maxSize) / 3 + 3) & ~3;
+        const jsonMinSize = 500;
+        maxSize = Math.max(maxSizeBase64, jsonMinSize);
```

Comment:
this needs to be documented I think, not trivial 😅

Reply:
done in https://github.com/RequestNetwork/requestNetwork/pull/837/commits/26f995a9be2391027a940208ef6b0a1091a08cd2